analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/HardcodedIpAddressBase.cs

```diff
@@ -100,6 +115,21 @@ private void CheckForHardcodedIpAddressesInStringInterpolation(SyntaxNodeAnalysi
             && !IPAddress.IsLoopback(ipAddress)
             && !ipAddress.GetAddressBytes().All(x => x == 0); // Nonroutable 0.0.0.0 or 0::0
 
+        private bool IsInDocumentationBlock(IPAddress address) =>
+            address.AddressFamily switch
+            {
+                AddressFamily.InterNetwork =>
+                    address.GetAddressBytes() is var ip
```

Comment:
Let's do the `GetAddressBytes()` before the switch in every case. It will simplify the logic a lot. And we query it everytime anyway